core/driver/sources/src/main/scala/org/virtuslab/ideprobe/dependencies/IntelliJZipResolver.scala

```diff
@@ -37,17 +38,10 @@ object IntelliJZipResolver extends IntelliJResolver {
       .resolver(artifact)
   }
 
-  def fromConfig(config: DependenciesConfig.IntelliJ): Seq[DependencyResolver[IntelliJVersion]] = {
-    val official = Seq(
-      IntelliJZipResolver.community,
-      AlternativeIntelliJZipResolver.community,
-      NightlyIntelliJZipResolver.community
-    )
-    val fromConfig = config.repositories
-      .flatMap(pattern =>
-        if (Set("official", "default").contains(pattern.toLowerCase)) official
-        else Seq(IntelliJPatternResolver(pattern).community)
-      )
-    if (fromConfig.isEmpty) official else fromConfig
+  // this method enables downloading .zip IntelliJ even if `probe.intellij.version.ext` value is not ".zip"
+  def fromConfig(config: DependenciesConfig.Resolvers): Seq[DependencyResolver[IntelliJVersion]] = {
```

Comment:
This sounds hacky. What would happen if we just don't replace it? Shouldn't it just fail if someone provides unsupported extension?

Reply:
My changes slightly change the logic used in `IntelliJProvider.from` so that:
- it no longer uses `IntelliJZipResolver.fromConfig`
- it uses `IntelliJResolver.fromConfig` instead - as the default method called when reading from config

So - `IntelliJZipResolver.fromConfig` will be a method that is not used by default when reading from config. It still could be used directly from code. It could be useful in some scenarios like below:
- if somebody uses `.dmg` as `probe.intellij.version.ext` (which will work after merging https://github.com/VirtusLab/ide-probe/pull/283 ) - but wants to use `.zip` in some specific scenario
- if somebody uses an Existing IntelliJ instead of `.zip` by default  - but wants to use `.zip` in some specific scenario

So, if someone provides unsopported extension - it will fail. But user's config will not go through `IntelliJZipResolver.fromConfig` but through `IntelliJResolver.fromConfig` (which is a more general method - could be used for more than `.zip` extension)

Reply:
Ah, I see. It is an extra method to load resolvers from config, but turn them into zip resolvers. Makes sense.